core/services/ocr3/plugins/ccip_integration_tests/ccipreader/ccipreader_test.go

```diff
@@ -0,0 +1,170 @@
+	reader := ccipreaderpkg.NewCCIPReader(lggr, contractReaders, contractWriters, chainD)
+
+	seqNums, err := reader.NextSeqNum(ctx, []cciptypes.ChainSelector{chainS1, chainS2, chainS3})
+	assert.NoError(t, err)
+	assert.Len(t, seqNums, 3)
+	assert.Equal(t, cciptypes.SeqNum(10), seqNums[0])
+	assert.Equal(t, cciptypes.SeqNum(20), seqNums[1])
+	assert.Equal(t, cciptypes.SeqNum(30), seqNums[2])
+}
+
+func TestCCIPReader_GasPrices(t *testing.T) {}
+
+func TestCCIPReader_Sync(t *testing.T) {}
+
+func testSetup(t *testing.T, ctx context.Context, destChain cciptypes.ChainSelector, onChainSeqNums map[cciptypes.ChainSelector]cciptypes.SeqNum) *testSetupData {
+	const chainID = 1337
+
+	// Generate a new key pair for the simulated account
+	privateKey, err := crypto.GenerateKey()
+	assert.NoError(t, err)
+	// Set up the genesis account with balance
+	blnc, ok := big.NewInt(0).SetString("999999999999999999999999999999999999", 10)
+	assert.True(t, ok)
+	alloc := map[common.Address]core.GenesisAccount{crypto.PubkeyToAddress(privateKey.PublicKey): {Balance: blnc}}
+	simulatedBackend := backends.NewSimulatedBackend(alloc, 0)
+	// Create a transactor
+
+	auth, err := bind.NewKeyedTransactorWithChainID(privateKey, big.NewInt(chainID))
+	assert.NoError(t, err)
+	auth.GasLimit = uint64(0)
+
+	// Deploy the contract
+	address, _, _, err := ccip_reader_tester.DeployCCIPReaderTester(auth, simulatedBackend)
+	assert.NoError(t, err)
+	simulatedBackend.Commit()
+
+	// Setup contract client
+	contract, err := ccip_reader_tester.NewCCIPReaderTester(address, simulatedBackend)
+	assert.NoError(t, err)
+
+	lggr := logger.TestLogger(t)
+	db := pgtest.NewSqlxDB(t)
+	lpOpts := logpoller.Opts{
+		PollPeriod:               time.Millisecond,
+		FinalityDepth:            0,
+		BackfillBatchSize:        10,
+		RpcBatchSize:             10,
+		KeepFinalizedBlocksDepth: 100000,
+	}
+	cl := client.NewSimulatedBackendClient(t, simulatedBackend, big.NewInt(0).SetUint64(uint64(destChain)))
+	headTracker := headtracker.NewSimulatedHeadTracker(cl, lpOpts.UseFinalityTag, lpOpts.FinalityDepth)
+	lp := logpoller.NewLogPoller(logpoller.NewORM(big.NewInt(0).SetUint64(uint64(destChain)), db, lggr),
+		cl,
+		lggr,
+		headTracker,
+		lpOpts,
+	)
+	assert.NoError(t, lp.Start(ctx))
+
+	for sourceChain, seqNum := range onChainSeqNums {
+		_, err := contract.SetSourceChainConfig(auth, uint64(sourceChain), ccip_reader_tester.CCIPReaderTesterSourceChainConfig{
+			IsEnabled: true,
+			MinSeqNr:  uint64(seqNum),
+		})
+		assert.NoError(t, err)
+		simulatedBackend.Commit()
+		scc, err := contract.GetSourceChainConfig(&bind.CallOpts{Context: ctx}, uint64(sourceChain))
+		assert.NoError(t, err)
+		assert.Equal(t, seqNum, cciptypes.SeqNum(scc.MinSeqNr))
+	}
+
+	return &testSetupData{
+		contractAddr: address,
+		contract:     contract,
+		sb:           simulatedBackend,
+		auth:         auth,
+		lp:           lp,
+		cl:           cl,
+	}
+}
+
+type testSetupData struct {
+	contractAddr common.Address
```

Comment:
Nit, for readability, address is already a duplicate as you have the contract and can do contract.Address()
```suggestion
	ccipTester     *ccip_reader_tester.CCIPReaderTester
```